asserts/device_asserts.go

```diff
@@ -105,11 +105,22 @@ func (mod *Model) checkConsistency(db RODatabase, acck *AccountKey) error {
 // sanity
 var _ consistencyChecker = (*Model)(nil)
 
+func checkBrandAndAuthority(a Assertion) error {
+	typeName := a.Type().Name
+	authorityID := a.AuthorityID()
+	brand := a.HeaderString("brand-id")
+	if brand != authorityID {
+		return fmt.Errorf("authority-id and brand-id must match, %s assertions are expected to be signed by the brand: %q != %q", typeName, authorityID, brand)
+	}
+	return nil
+}
+
 var modelMandatory = []string{"core", "architecture", "gadget", "kernel", "store", "class"}
 
 func assembleModel(assert assertionBase) (Assertion, error) {
-	if assert.headers["brand-id"] != assert.headers["authority-id"] {
-		return nil, fmt.Errorf("authority-id and brand-id must match, model assertions are expected to be signed by the brand: %q != %q", assert.headers["authority-id"], assert.headers["brand-id"])
+	err := checkBrandAndAuthority(&assert)
+	if err != nil {
```

Comment:
@mvo5 I think you had this nitpick already a couple of times, should I really switch to that style,  it seems not to be really in my fingers